tokio/tests/signal.rs

```diff
@@ -0,0 +1,169 @@
+#![cfg(unix)]
+#![warn(rust_2018_idioms)]
+
+use tokio::prelude::*;
+use tokio::runtime::current_thread::Runtime;
+use tokio::signal;
+use tokio::signal::unix::{signal, SignalKind};
+use tokio::sync::oneshot;
+use tokio_test::assert_ok;
+
+use futures::future;
+use std::sync::mpsc::channel;
+use std::thread;
+
+#[tokio::test]
+async fn signal_usr1() {
```

Comment:
The `signal` implementation uses a bunch of global structures/registrations to move things along, and the tests were originally written such that each test would run in a fresh process (to avoid having multiple tests interfere with each other, especially when all the tests run in parallel in the same process).

This proved to be especially important when certain deadlock/starvation issues were discovered, which would have been hidden if the tests were running together (basically the executor of one test could inadvertently end up driving another test which would have starved).

I think there is some merit in keeping the signal tests in separate integration-style tests as proofing from any bugs slipping through. Thoughts?

Reply:
Yes, I wasn't sure. Would it help if tests are ran single threaded? Otherwise, I can split them up again.

Reply:
Thanks for splitting these out!